website/sales/views.py

```diff
@@ -0,0 +1,38 @@
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import get_object_or_404, redirect, render
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext_lazy as _
+from django.views import View
+
+from sales import services
+from sales.models.order import Order
+
+
+@method_decorator(login_required, name="dispatch")
+class OrderPaymentView(View):
+    def get(self, request, *args, **kwargs):
+        order = get_object_or_404(Order, pk=kwargs["pk"])
+        if order.payment:
+            messages.warning(request, _("This order was already paid for."))
+            return redirect("index")
```

Comment:
I think it would be neat to render a different page instead of redirecting to the index. That way the error will be the most important thing on the page.

Reply:
I think it isn't necessary to do this, as it is a hypothetical situation that should never happen in practice if people don't try to mess with the system

Reply:
That is a good point.